### PR TITLE
Add Means of Tracking Origin of Connection's CTF

### PR DESCRIPTION
**Pull request by bska**
This commit adds a new type (Connection::CTFSource), and new data member of this type (`Connection::m_ctfsource`, initialised in the constructor) that tracks the source of the connection'st transmissibility factor (`Connection::m_CF`).  The primary user of this information is the restart file writing code which needs to know if the connection transmissbility factor is assigned from the input file (e.g., directly from the `COMPDAT` keyword) or if the value is calculated from other information.

Add a convenience predicate function,
```C++
bool Connection::ctfAssignedFromInput() const
```
that returns true if the connection transmissibility factor source indeed is the `COMPDAT` keyword.

Update Connection constructor callers accordingly.

---

This PR introduces helper functionality for parts of #1195.

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
This looks fine - but just one thing which I hope you can consider: Other places in the opm-common code we have the concept of *defaulted* - which signifies that a quantity has a default value. The default value can either just originate in a `'*'` from the input deck, or it can be a more complex calculation as for the saturation functions.

I get a feeling that this could be a meaningful approach also for this particular variable; i.e. either the connection factor has been explicitly entered in the deck - or it has been defaulted ("calculated from other informat"). Not that one is inherently better than the other, but since the *defaulted* concept already is in use I would suggest that?

File: opm/parser/eclipse/EclipseState/Schedule/Well/Connection.hpp [hunk 1]
```diff
@@ -67,7 +67,12 @@ namespace Opm {
         static const std::string Order2String( Order enumValue );
         static Order OrderFromString(const std::string& comporderStringValue);
 
+        enum class CTFSource {
+            AssignedFromInput,
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Schedule/Well/Connection.hpp]: This could be:
```
enum class CTFStatus {
     DeckValue,
     Defaulted,
};
```

**Comment by bska** [comment]
I have renamed the `enum` to `CTFKind` with enumerations `DeckValue` and `Defaulted`.

**Comment by bska** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
> I have renamed the enum to CTFKind with enumerations DeckValue and Defaulted.

Cool - thank you. Merging

**Pull request merged**